strings/strings.c

```diff
@@ -20,7 +25,16 @@ int string_length(char *s)
 */
 char *reverse_string(char *rv, char *s)
 {
-
+    int length = string_length(s);
+    int current = 0;
+
+    while (length >= 0) {
```

Comment:
In a lot of places in this project, you're using `-=` and `+=` to decrement and increment. This is how we had to do it in python, but we can just use `--` and `++` in C - like Javascript.